ballerina/engine.bal

```diff
@@ -97,22 +94,28 @@ isolated class Engine {
         return getOutputObjectFromErrorDetail(errorDetail);
     }
 
-    isolated function validateDocument(ParseResult parseResult, map<json>? variables)
-    returns OutputObject|parser:DocumentNode {
-        ErrorDetail[]|NodeModifierContext validationResult = self.parallellyValidateDocument(parseResult, variables);
+    isolated function validateDocument(ParseResult parseResult, string? operationName, map<json>? variables)
+    returns OutputObject|parser:OperationNode {
+        ErrorDetail[]|NodeModifierContext validationResult =
+            self.parallellyValidateDocument(parseResult, operationName, variables);
         if validationResult is ErrorDetail[] {
             return getOutputObjectFromErrorDetail(validationResult);
         } else {
             DocumentNodeModifierVisitor documentNodeModifierVisitor = new (validationResult);
             parseResult.document.accept(documentNodeModifierVisitor);
-            return documentNodeModifierVisitor.getDocumentNode();
+            return documentNodeModifierVisitor.getDocumentNode().getOperations()[0];
```

Comment:
I think returning the `OperationNode` from the `validateDocument` method is kinda incorrect. Shall return the `DocumentNode` here and access the operation node in the `validate` method? When accessing, better to put a comment about why we directly access the 0th operation node, because, at the first glance, it is not easily understandable why we use the 0th operation every time. Took some time for me here as well.

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerina-graphql/pull/1297/commits/2f3f897d6bbbabd7457d6cfe38dead3f347cd818